src/extract_sf.c

```diff
@@ -240,7 +243,14 @@ int run(char *in_filename, bool use_fmt, int nevn, int run_no) {
     }
 
     // Create output file.
-    TFile *f_out = TFile::Open(Form("../root_io/sf_study_%06d.root", run_no), "RECREATE");
+    char*  out_filename = (char *) malloc(128 * sizeof(char));
+    if(!use_simul){
+        sprintf(out_filename, "../root_io/sf_study_banks_%06d.root", run_no);
+    } else{
+        sprintf(out_filename, "sf_study_%s", in_filename);
```

Comment:
This filename is wrong too, it's not saved to the correct directory and it has no extension.

Reply:
Perhaps instead of adding a lot of if-else statement it would be easier to assign an arbitrary run number to simulations and work from there? It's a bit dirty but as long as it's correctly explained to the user it seems like a much more maintainable solution.

Reply:
Fixed the extension issue.
I like the idea of working with a specific run number for simuls. I have two possible numbers in mind:
1.- 11 : In gemc there are some options taken from certain runs. 11 is one of them. I really doubt that we'll use this program to treat that same run number; therefore, we could use that.
2.- Something weird but identifiable like "101010". I doubt there is a usable run with that same number.

Reply:
Check in the software developer documentation, I'm pretty confident there are some run numbers reserved for simulation. If there's no info I can ask around at the lab on Monday.

Reply:
Please ask :D

Reply:
Anyways, I was thinking that for simuls we could use the following format:
6 digits : 999XYZ
Where XYZ is the energy of the simul multiplied by ten. This, in case a simul with fractionary energy is used.
Examples:
Ebeam = 11    Gev  -> Run no : 999110
Ebeam = 10.6 GeV -> Run no : 999106
Ebeam = 22    GeV -> Run no : 999220

Reply:
That's not a bad idea. Let's use that standard. If we eventually find out there's another one at the lab we can just switch without much effort.